jaggery-modules/taxonomy/module/scripts/taxonomy/taxonomy.js

```diff
@@ -45,8 +47,21 @@ var GovernanceAPITree = org.wso2.carbon.governance.api.util.TaxonomyTreeAPI;
      * @return {Object}  Categories list as paths, which admin defined.
      */
     taxonomy.getTaxa = function () {
-        return JSON.parse(GovernanceCommon.getPathCategories());
+        if (!session.get(TAXA)) {
+            return putInSession();
+        } else {
+            if (session.get(LAST_MODIFIED_TIME) != TaxonomyService.getLastModifiedTime()) {
+                return putInSession();
+            }
+            return session.get(TAXA);
+        }
+
     };
 
+    function putInSession() {
```

Comment:
**Important** We should put this in the application context as the Taxonomy is unlikely to change on a per session basis